content/ADR-145-pick-realm-v2.md

```diff
@@ -3,49 +3,69 @@ layout: adr
 adr: 145
 title: Pick Realm Algorithm V2
 date: 2022-11-28
-status: Draft
+status: Living
 type: RFC
 spdx-license: CC0-1.0
 slug: /adr/TEMPLATE
 authors:
   - agusaldasoro
+  - aleortega
 ---
 
 ## Abstract
 
-This document describes the changes needed in the realm picking algorithm to take into account the resources of the computer and the catalyst version.
+The purpose of this document is to outline the necessary modifications to improve the realm picking algorithm, both in terms of its functionality and implementation.
 
-The logic on the realm scoring algorithm should be improved to take into consideration more variables in order for a server to be more eligible. As with the current model, any added variable needs to be modifiable through Feature Flags so that they can be changed in case of need without a release or a reboot of the server.
+With regard to functionality, the algorithm will be updated to incorporate additional variables in the Catalyst selection process through the addition of new rules. As for the implementation, the software will be relocated to the server-side to accommodate multiple client implementations.
 
 ## Context, Reach & Prioritization
 
-Some aspects that should be considered:
+The realm-picking algorithm utilizes a predefined set of rules called `variants` to dynamically switch to the optimal approach based on the platform context through configuration files. However, it has been identified that this algorithm lacks certain capabilities to address potential contexts that Decentraland may encounter. To enhance the selection algorithm, the following improvements are intended:
 
-- Resources: if a server has either CPU or memory at 90%, don't add more load to that server.
-- Catalyst version: if the version of the node is behind the network, it should be less eligible.
+- Addition of new rules:
+  - Overloaded Catalyst: Servers with CPU or memory usage exceeding 90%, or reaching maximum user capacity, will be excluded from selection.
+  - Catalyst Version: Nodes with a version lower than the minimum required version (_as specified in the configuration_) will be excluded from selection. This will allow us to set minimum required version for each service running within a Catalyst.
+  - Catalyst Force: Possibility to forcibly select a Catalyst.
+- Creation of multiple default rule sets (known as _variants_) that can be easily switched to use different techniques for realm picking.
+- Relocating the algorithm to the server-side to accommodate multiple client implementations without losing any capabilities.
 
+From now on, the rules will be referred to as `links` and `filters`. Filters will be used to remove nodes that do not meet specific criteria, while links will be used to assign scores to nodes to determine the best one for redirecting users.
 
 ## Solution Space Exploration
```

Comment:
It's not clear to me where will be the configuration file live, also who will be in charge of it? is it the client's responsibility? Is it a configuration of the centralized server/lambda/worker?